src/app/beer_garden/requests.py

```diff
@@ -677,6 +678,36 @@ def start_request(request_id: str = None, request: Request = None) -> Request:
     return request
 
 
+@publish_event(Events.REQUEST_UPDATED)
+def update_request(
+    request_id: str = None,
+    request: Request = None,
+    new_values=None,
+) -> Request:
+    if new_values is None:
+        new_values = {}
+    request = request or db.query_unique(Request, raise_missing=True, id=request_id)
+    if "expiration_date" in new_values:
+        expiration_date = new_values.pop("expiration_date", None)
+        parent = find_parent(request)
+        parent.expiration_date = expiration_date
+        parent = db.update(parent)
+        if parent.id == request_id:
+            request = parent
+    for key in new_values.keys():
+        setattr(request, key, new_values[key])
+    if new_values:
+        request = db.update(request)
+    return request
+
+
+def find_parent(request):
```

Comment:
This should be named something else. Returning the request passed should not be the behavior of a function called "find_parent."